rust/operator-binary/src/controller.rs

```diff
@@ -703,7 +728,35 @@ fn build_server_rolegroup_daemonset(
             ..Probe::default()
         });
 
-    let mut pb = PodBuilder::new();
+    if let Some(user_info) = &opa.spec.cluster_config.user_info {
+        cb_user_info_fetcher
+            .image(user_info_fetcher_image)
```

Comment:
The UIF comes from the _operator_ image, not the product image.

Reply:
While you could argue that the product image would be more correct.. the product image build process is enough of a mess as it is, and the strong dependency here is ultimately in making sure that the UIF and operator are compatible.

Reply:
This also isn't the only case of this, https://github.com/stackabletech/opa-operator/tree/spike/bundle-builder-v2 gives the bundle builder the same treatment.

Reply:
There's also the DX side, testing repo-local changes are a lot less painful (especially when covered by tilt) than cross-repo changes.

Reply:
I agree that DX becomes a little more difficult now, but I think that could be good reason to move UIF out into its own repo with its own lifecycle (much like the bundle builder).

It is deployed by the operator, but it really isn't a component of the operator so it feels like it is only hanging around here because of DX.

Reply:
> I agree that DX becomes a little more difficult now, but I think that could be good reason to move UIF out into its own repo with its own lifecycle (much like the bundle builder).

How would DX be improved by that? Taking the bundle builder as a reference, you'd need to:

1. Make your change
2. Commit and push your change
3. Note the commit hash
4. Open the product Dockerfile (hope you have the right version of docker-images checked out already!)
5. Paste the commit hash into the Dockerfile
6. `bake`
7. Go make a coffee while you wait, because Docker's build caching is a sad joke
8. Go back to step 6, because you forgot to give the image a unique tag
9. Manually redeploy the `OpaCluster` with the new image version

And that's not even considering making sure that nightly operator versions match up with the right UIF versions, or that you now _also_ have to consider UIF config versioning, since you can't guarantee that the UIF uses the same version of the scheme as the operator anymore.

Reply:
I don't have a comprehensive answer on how DX could be improved, but I'm sure it could in the same way other things have been improved. Naivly, I would say that when doing user-info-fetcher development, Tilt can go and supply the correct image for deployment by the opa-operator.

But in this case, we have agreed to keep it where it is, so I believe we just need to revert a change and this can be resolved.

Reply:
Reverted in 6dcc44cad25df9c2a797727efa84a2e4bd637af7 and the additional `.image` call added at @sbernauer's request.

Reply:
Thanks!